### PR TITLE
<Checkbox /> - Clean & Improve comp

### PR DESCRIPTION
**Pull request by danielstro**
This is the first of several changes regarding <Checkbox />. 
Mostly cleaning code and some cosmetic changes, but also changing the checked state to use enum and removing this.id.
For more information see this issue #259

### CODE REVIEW

File: packages/wix-ui-core/src/components/Checkbox/CheckboxStyle.st.css [hunk 1]
```diff
@@ -39,8 +39,8 @@
     border: value(borderDefault);
 }
 
-.root:checked::box,
-.root:indeterminate::box {
+.root:checked(true)::box,
```
Comment (alisey) [inline packages/wix-ui-core/src/components/Checkbox/CheckboxStyle.st.css]: We're losing the ability to simply write `.myCheckbox:checked`. I was dealing with a similar issue in the toggle switch yesterday, I wanted to have `:focus` and `:focus(mouse)`, but it seems like Stylable doesn't support this at the moment.

Comment (idoros) [reply packages/wix-ui-core/src/components/Checkbox/CheckboxStyle.st.css]: I agree, I know this is going back from the previous review, but its better as 2 states: checked & indeterminate

File: packages/wix-ui-core/src/components/Checkbox/Checkbox.spec.tsx [hunk 2]
```diff
@@ -95,7 +81,7 @@ describe('Checkbox', () => {
       expect(document.activeElement).toBe(checkbox.input());
 
     } else {
-      console.warn(// tslint:disable-line no-console
+      console.error(// tslint:disable-line no-console
```
Comment (idoros) [inline packages/wix-ui-core/src/components/Checkbox/Checkbox.spec.tsx]: In which scenario are the test run without focus support?

Comment (danielstro) [reply packages/wix-ui-core/src/components/Checkbox/Checkbox.spec.tsx]: I think this is a leftover from stylable components, I think the tests were flakey without this. I will remove it and try to see if this affects the tests.

**Pull request merged**